ow_lander/src/ow_lander/actions.py

```diff
@@ -72,6 +73,54 @@ def _compute_workspace_shoulder_yaw(x, y):
     yaw += math.asin(l / h)
     _assert_shou_yaw_in_range(yaw)
     return yaw
+    
+class _arm_faults_internal_messages_publish:
+    """Publish the arm failure messages to rostopic 'arm_faults_internal'"""
+    def __init__(self, err):
+      self.err = err
+      self.arm_faults_internal_pub = rospy.Publisher('arm_faults_internal', ArmFaultsStatus, queue_size = 10, latch = True)
+
+      """ error messages translate"""
+      if self.err == 'NONE':
+          self.err_flag = 0
+      elif self.err == 'HARDWARD':
+          self.err_flag = 1
+      elif self.err == 'TRAJECTORY_GENERATION':
+          self.err_flag = 2
+      elif self.err == 'COLLISION':
+          self.err_flag = 4
+      elif self.err == 'E_STOP':
+          self.err_flag = 8
+      elif self.err == 'POSITION_LIMIT':
+          self.err_flag = 16
+      elif self.err == 'JOINT_TORQUE_LIMIT':
+          self.err_flag = 32
+      elif self.err == 'VELOCITY_LIMIT':
+          self.err_flag = 64
+      elif self.err == 'NO_FORCE_DATA':
+          self.err_flag = 128
+      elif self.err == 'FORCE_TORQUE_LIMIT':
+          self.err_flag = 256
+      else:
+          rospy.loginfo('Arm faults status message is invalid')
+          return
+    
+      self.message_received = False
+
+      rospy.loginfo(self.err + ' error message sending now')
+
+      # To ensure that the subscriber has receives the message, keep publish until all set bits on
+      # self.err_flag also in msg.value now.
+      while not rospy.is_shutdown() and not self.message_received:
+        if self.arm_faults_internal_pub.get_num_connections() > 0:
+          self.arm_faults_internal_pub.publish(value = self.err_flag)
+          self.message_received = True
```

Comment:
I believe ROS already has ways to ensure messages get received, and considering I never see a message local to the machine dropped, I think this loop is unnecessary.